src/main/java/org/myrobotlab/framework/interfaces/ServiceInterface.java

```diff
@@ -172,59 +167,7 @@
 
   public int getCreationOrder();
 
-  static public LinkedHashMap<String, ServiceConfig> getDefault(String name, String type) {
-    LinkedHashMap<String, ServiceConfig> config = new LinkedHashMap<>();
-    
-    String simpleTypeName = null;
-    if (type.contains(".")) {
-      simpleTypeName = type.substring(type.lastIndexOf(".")+1);
-    } else {
-      simpleTypeName = type;
-    }
-    
-    try {
-
-      String metaClass = "org.myrobotlab.service.meta." + simpleTypeName + "Meta";
-
-      Class<?> clazz = Class.forName(metaClass);
-      Method method = clazz.getMethod("getDefault", String.class);
-
-      // I chose "non"-static method for getDefault - because Java has
-      // an irritating rule of not allowing static overloads and abstracts
-      config = (LinkedHashMap<String, ServiceConfig>) method.invoke(null, name);
-
-      if (config == null || config.keySet().size() == 0) {
-        log.warn("{} does not currently have any default configurations", metaClass);
-      }
-    } catch (NoSuchMethodException em) {
-
-      try {
-
-        log.info("{} of type {} does not have a getDefault", name, type);
-
-        String configClass = "org.myrobotlab.service.config." + simpleTypeName + "Config";
-
-        Class<?> clazz = Class.forName(configClass);
-
-        // create new instance
-        Constructor<?> ctor = clazz.getConstructor();
-        ServiceConfig configObject = (ServiceConfig) ctor.newInstance();
-
-        // I chose "non"-static method for getDefault - because Java has
-        // an irritating rule of not allowing static overloads and abstracts
-        config.put(name, configObject);
-
-      } catch (Exception e) {
-        log.warn("{} of type {} does not have a Config object - creating default service config", name, type);
-        ServiceConfig serviceConfig = new ServiceConfig();
-        serviceConfig.type = type;
-        config.put(name, serviceConfig);
-      }
-
-    } catch (Exception e) {
-      log.error("ServiceConfig.getDefault({},{}) threw", name, type, e);
-    }
-    return config;
-  }
+  public Plan setPlan(Plan plan);

```

Comment:
add javadoc please.  it's not clear when this should be called, or that once you call it, you can't call it again..

Reply:
this method is going to go away